contracts/SpokeV3PoolPeriphery.sol

```diff
@@ -13,26 +13,29 @@ import { WETH9Interface } from "./external/interfaces/WETH9Interface.sol";
 
 /**
  * @title SpokePoolV3Periphery
- * @notice Contract for performing more complex interactions with an AcrossV3 spoke pool deployment.
- * @dev Variables which may be immutable are not marked as immutable, nor defined in the constructor, so that this contract may be deployed deterministically.
+ * @notice Contract for performing more complex interactions with an Across spoke pool deployment.
+ * @dev Variables which may be immutable are not marked as immutable, nor defined in the constructor, so that this
+ * contract may be deployed deterministically at the same address across different networks.
  * @custom:security-contact bugs@across.to
  */
 contract SpokePoolV3Periphery is Lockable, MultiCaller {
     using SafeERC20 for IERC20;
     using Address for address;
 
     // This contract performs a low level call with arbirary data to an external contract. This is a large attack
-    // surface and we should whitelist which function selectors are allowed to be called on the exchange.
-    mapping(bytes4 => bool) public allowedSelectors;
+    // surface and we should whitelist which function selectors are allowed to be called on which exchange.
+    mapping(address => mapping(bytes4 => bool)) public allowedSelectors;
+
+    struct WhitelistedExchanges {
+        address exchange;
+        bytes4[] allowedSelectors;
+    }
 
     // Across SpokePool we'll submit deposits to with acrossInputToken as the input token.
     V3SpokePoolInterface public spokePool;
 
-    // Exchange address or router where the swapping will happen.
-    address public exchange;
-
     // Wrapped native token contract address.
-    WETH9Interface internal wrappedNativeToken;
+    WETH9Interface public wrappedNativeToken;
```

Comment:
any harm in making this public? seems potentially useful for deployer to verify

Reply:
No harm. I made it internal only to save gas, but I agree it's annoying to have to check the `initialize` transaction to verify a correct value over just making it public.

Reply:
how does it save gas?

Reply:
It's very small, but public variables cost more than private/internal variables on deployment.

Reply:
ah ok i thought you were referencing gas costs on post deployment function calls, got it. we can live with that